JPetTask/JPetTask.cpp

```diff
@@ -32,12 +32,12 @@ std::string JPetTask::getName() const
   return fName;
 }
 
-void JPetTask::setSubTask(std::unique_ptr<JPetTaskInterface> subTask)
+void JPetTask::addSubTask(std::unique_ptr<JPetTaskInterface> subTask)
 {
-  fSubTask = std::move(subTask);
+  fSubTasks.push_back(std::move(subTask));
 }
 
-JPetTaskInterface* JPetTask::getSubTask() const
+const std::vector<std::unique_ptr<JPetTaskInterface>>* JPetTask::getSubTasks() const
```

Comment:
why not to return a vector of raw pointers ? If you return a vector of unique_ptr you are still telling that the task is not an owner of subtasks anymore, it transfers all the ownership to the one that calls it.
The best would be to add a test with a one task class that has at least two subclasses  and check how it behaves.

Reply:
I'm returning pointer to vector of unique_ptr, not vector itself, but if vector of raw pointers is better I can change it. I don't know what will happen when somebody will cast pointer to reference, so I will change it to vector of raw pointers